src/res/KM_ResKeys.pas

```diff
@@ -68,6 +91,31 @@ constructor TKMKeyLibrary.Create;
   fKeymapPath := (ExeDir + 'keys.keymap');
 
   LoadKeymapFile;
+
+  for I := 0 to KEYMAP_COUNT - 1 do
+  begin
+    fFuncs[I].TextId := KEY_FUNC_TX[I];
+
+    if (I in [0..3, 24..26, 40..44]) then
+      fFuncs[I].Area := kaCommon
+    else if (I in [4..23, 27..39]) then
+      fFuncs[I].Area := kaGame
+    else
+      fFuncs[I].Area := kaMapEdit;
+
+    fFuncs[I].Id := I;
```

Comment:
Now there's a question, why we need Id that is always the same as the element index in array?
fFuncs[I].Id := I;